### PR TITLE
Pipenv bump v2022.4.8

### PR DESCRIPTION
**Pull request by oz123**
The upstream version removed some bundled dependencies (first, doctopt, yarg and pipreqs).
Gentoo version removed bundled click and iso8601.

Bug: https://bugs.gentoo.org/717666

### CODE REVIEW

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	done
+
+	find  ./pipenv/vendor -regextype posix-extended -regex '.*cached[_-]property.*' -prune -exec rm -rvf {} + || die
+
+	# not actually used by pipenv, but included in pipenv	
+	rm -vR "${S}/${PN}/vendor/wheel/" || die
+}
+
+python_test() {
+	pytest -vvv -x -m "not cli and not needs_internet" tests/unit/ || die
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:92]:
```suggestion
	epytest -m "not cli and not needs_internet" tests/unit/
```

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+BDEPEND="
+	${RDEPEND}
+	test? (
+		dev-python/flaky[${PYTHON_USEDEP}]
+		dev-python/mock[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytz[${PYTHON_USEDEP}]
+	)
+"
+
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:55]: Please use here `distutils_enable_tests pytest`, in which case you can remove `pytest` from test deps, and remove the lines:
```
IUSE="test"
RESTRICT="!test? ( test )"
```

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+S="${WORKDIR}"/${PN}-${MY_PV}
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~riscv ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	${PYTHON_DEPS}
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:24]:
```suggestion
```
Auto added by `distutils-r1` eclass

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,93 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DISTUTILS_USE_SETUPTOOLS=rdepend
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:6]: Please use newer [PEP517 syntax](https://projects.gentoo.org/python/guide/distutils.html#basic-use-pep-517-mode)

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+# IMPORTANT: The following sed command patches the vendor direcotry
+# in the pipenv source. Attempts to simply bump the version of the
+# package without checking that it works is likely to fail
+# The vendored packages should eventually all be removed
+# see: https://bugs.gentoo.org/717666
+src_prepare() {
+
+	local jobs=$(makeopts_jobs)
+	local packages=( attr cerberus cached_property click colorama idna iso8601 pexpect dateutil requests toml tomli urllib3 zipp )
+	for pkgName in ${packages[@]}; do
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:65]: please add before this line `local pkgName`

**Review by arthurzam — CHANGES_REQUESTED** [review]
Various changes from partial review

File: dev-python/pipenv/pipenv-2022.4.8.ebuild (hunk 4 above)
Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild:6]: Is this instead of `DISTUTILS_USE_SETUPTOOLS=rdepend` or in addition?

Comment (arthurzam) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild:6]: instead

Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild]: Thanks!

File: dev-python/pipenv/pipenv-2022.4.8.ebuild (hunk 5 above)
Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild:63]: done.

File: dev-python/pipenv/pipenv-2022.4.8.ebuild (hunk 1 above)
Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild]: done.

**Comment by oz123** [comment]
@arthurzam thanks for the review. I implemented all the things you sugessted.

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		find ./ -type f -print0 | \
+			xargs --max-procs="${jobs}" --null \
+			sed --in-place \
+				-e 's/from pipenv.vendor import '"${pkgName}"'/import '"${pkgName}"'/g' \
+				-e 's/from pipenv.vendor.'"${pkgName}"'\(.*\) import \(\w*\)/from '"${pkgName}"'\1 import \2/g' \
+				-e 's/import pipenv.vendor.'"${pkgName}"' as '"${pkgName}"'/import '"${pkgName}"'/g' \
+				-e 's/from .vendor import '"${pkgName}"'/import '"${pkgName}"'/g'
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:70]: If I'm reading the seds correctly, I think we can simplify them using double quotes. Also, lets add the `|| die` and use `find -exec`.

```suggestion
		find . -type f -exec sed -i {} \
				-e "s/from pipenv.vendor import ${pkgName}/import ${pkgName}/g" \
				-e "s/from pipenv.vendor.${pkgName}\(.*\) import \(\w*\)/from ${pkgName}\1 import \2/g" \
				-e "s/import pipenv.vendor.${pkgName} as ${pkgName}/import ${pkgName}/g" \
				-e "s/from .vendor import ${pkgName}/import ${pkgName}/g" + || die "Failed to sed for ${pkgName}"
```

And then you can remove the `assert` after the loop

---

But please check that I did this correctly 😄

File: dev-python/pipenv/pipenv-2022.4.8.ebuild [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	done
+	assert "Failed to sed sources"
+
+	distutils-r1_src_prepare
+
+	# remove vendored versions
+	for pkgName in ${packages[@]}; do
+	    find  ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName}$" -prune -exec rm -rvf {} + || die
+		# package names can be foo-bar, their module will be however foo_bar
+		find  ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName/_/-}" -prune -exec rm -rvf {} + || die
```
Comment (arthurzam) [inline dev-python/pipenv/pipenv-2022.4.8.ebuild:80]:
```suggestion
		find ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName}$" -prune -exec rm -rvf {} + || die
		# package names can be foo-bar, their module will be however foo_bar
		find ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName/_/-}" -prune -exec rm -rvf {} + || die
```

**Review by arthurzam — CHANGES_REQUESTED** [review]
I hope the last changes that I request. Great work!

Also, please review all the indentations in this file, and check they are using TABs.

File: dev-python/pipenv/pipenv-2022.4.8.ebuild (hunk 6 above)
Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild]: Once again, thanks for the review! Much appreciated.

File: dev-python/pipenv/pipenv-2022.4.8.ebuild (hunk 7 above)
Comment (oz123) [reply dev-python/pipenv/pipenv-2022.4.8.ebuild]: added the changes you suggested.

**Review by arthurzam — APPROVED** [review]
Sorry for the delay 😄 
Looks good, thanks!

**Pull request closed without merging**